In the 423/435 stream devices, dev.copy() was moved back to the main process.

santorofer opened the pull request:
The 423/435 streaming devices contain a call to copy the tree (`dev.copy()`) inside the MDSWorker's new thread. 
This breaks the code when executing it from an mdsip service (i.e. when dispatching the INIT action).
Moving that statement back to the main process thread fixes the issue.

santorofer committed:
Moved the tree copy (dev.copy()) from the MDSWorker thread back to the main process thread.

santorofer committed:
Moved the creation of chans[] and decim[] to MDSWorker thread.

santorofer committed:
For 423 devices, dev.copy() has also been moved

WhoBrokeTheBuild commented on pydevices/HtsDevices/acq2106_435st.py:
```diff
@@ -188,9 +188,9 @@ class MDSWorker(threading.Thread):
+            self.dev = dev.copy()
 
-            self.nchans     = self.dev.sites*32
+            self.nchans     = self.dev.sites * 32
```
Why 32? Can we put that in a constant or add a comment?

WhoBrokeTheBuild commented on pydevices/HtsDevices/acq2106_435st.py:
```diff
@@ -469,12 +466,12 @@ def init(self, resampling = False, armed_by_transient = False):
-        self.chans = []
-        nchans = uut.nchan()
+        chans = []
+        nchans = self.sites * 32
```
Same question as above, maybe do a static constant in the class

santorofer committed:
Converted self.chans and self.decim to chans and decim.

santorofer committed:
Added static variable NUM_CHANS_PER_SITE

WhoBrokeTheBuild commented on pydevices/HtsDevices/acq2106_423st.py:
```diff
@@ -159,15 +159,17 @@ class _ACQ2106_423ST(MDSplus.Device):
+            self.dev = dev.copy()
 
-            self.nchans = self.dev.sites*32
+            self.nchans = self.dev.sites * NUM_CHANS_PER_SITE
```
You still need to use `self.NUM_CHANS_PER_SITE`

WhoBrokeTheBuild commented on pydevices/HtsDevices/acq2106_435st.py:
```diff
@@ -182,15 +182,17 @@ class _ACQ2106_435ST(MDSplus.Device):
+            self.dev = dev.copy()
 
-            self.nchans     = self.dev.sites*32
+            self.nchans     = self.dev.sites * NUM_CHANS_PER_SITE
```
You will need to use `self.dev.NUM_CHANS_PER_SITE`

santorofer committed:
Corrected the calls to the static variable NUM_CHANS_PER_SITE

santorofer committed:
Create a tuple containing an instant of the tree to be used in the MDSWorker thread (instead of using .copy()

WhoBrokeTheBuild commented:
We should switch to passing the shot ID and tree name to the child thread, and having it create its own Tree object instead of calling .copy()

WhoBrokeTheBuild commented on pydevices/HtsDevices/acq2106_423st.py:
```diff
@@ -159,15 +159,20 @@ class _ACQ2106_423ST(MDSplus.Device):
             super(_ACQ2106_423ST.MDSWorker, self).__init__(name=dev.path)
 
+            # Tuple designed to bring a copy of the tree to the MDSWorker thread
+            self.info = (dev.tree.name, dev.tree.shot, dev.path)
```
Originally this was made to emulate the to_info and from_info functions that I made for Josh, but we've replaced those with just multiple variables, so I think we should do the same here.
```
self.tree = dev.tree.name
self.shot = dev.tree.shot
self.path = dev.path

...

tree = MDSplus.Tree(self.tree, self.shot)
self.dev = tree.getNode(self.path)
```

santorofer replied:
Ok, this looks good to me too. I'll make those changes then.

santorofer committed:
Break tuple into 3 variables

Replace info tuple with tree, shot and path to bring a copy of the tree to the worker thread.
